lib/system/database/definitions/mysql.rb

```diff
@@ -471,7 +471,9 @@
 
   trigger 'event_store_events' do
     <<~SQL
-      SET NEW.tenant_id = NEW.provider_id;
+      IF NEW.provider_id <> master_id THEN
```

Comment:
I sent a message earlier somewhere that in old installations, master account has  `tenant_id == nil`. In such an installation the above condition doesn't matter because master owned events will still have it also `nil`.

But in newer installations, the master account has `tenant_id == account.id`. In such situations events would have a non-null tenant_id and the call to create a new provider account would fail because events are being generated.

There is some info in d9c51dc8687bc85fac526d23fdb23c8c4c9d4c5f